fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/data/LoanSchedulePeriodData.java

```diff
@@ -402,6 +403,49 @@ private BigDecimal defaultToZeroIfNull(final BigDecimal possibleNullValue) {
         return value;
     }
 
+    public BigDecimal getRatePerDay() {
+        return getInterestOriginalDue().divide(BigDecimal.valueOf(getDaysInPeriod()), MoneyHelper.getMathContext());
+    }
+
+    public BigDecimal getInterestDueUntilDate(final LocalDate businessDate) {
+        if (getComplete()) {
+            return BigDecimal.ZERO;
+        } else {
+            if (businessDate.compareTo(getDueDate()) >= 0) { // Period is matured
+                return getInterestOriginalDue();
```

Comment:
This is incorrect. We want to know what interest was not yet paid and not yet due. If we are on or over the due date, it cannot be not yet due, so it must be 0.

Reply:
Done! solved